src/Native/System.Net.Security.Native/pal_gssapi.cpp

```diff
@@ -0,0 +1,257 @@
+}
+
+extern "C" uint32_t NetSecurity_DisplayStatus(uint32_t* minorStatus,
+                                              uint32_t statusValue,
+                                              int32_t isGssMechCode,
+                                              gss_buffer_t* outBufferHandle,
+                                              uint32_t* statusLength)
+{
+    assert(minorStatus != nullptr);
+    assert(outBufferHandle != nullptr);
+    assert(statusLength != nullptr);
+    assert(isGssMechCode == 0 || isGssMechCode == 1);
+    int statusType = isGssMechCode ? GSS_C_MECH_CODE : GSS_C_GSS_CODE;
+    *outBufferHandle = new gss_buffer_desc();
+    uint32_t majorStatus =
+        gss_display_status(minorStatus, statusValue, statusType, GSS_C_NO_OID, nullptr, *outBufferHandle);
+    NetSecurity_HandleError(majorStatus, outBufferHandle, statusLength);
+    return majorStatus;
+}
+
+extern "C" uint32_t
+NetSecurity_ImportUserName(uint32_t* minorStatus, char* inputName, uint32_t inputNameLen, gss_name_t* outputName)
+{
+    assert(outputName != nullptr);
+    assert(minorStatus != nullptr);
+    assert(inputName != nullptr);
+    assert(outputName != nullptr);
+    gss_buffer_desc inputNameBuffer{.length = inputNameLen, .value = inputName};
+    gss_OID nameType = const_cast<gss_OID>(GSS_C_NT_USER_NAME);
+    return gss_import_name(minorStatus, &inputNameBuffer, nameType, outputName);
+}
+
+extern "C" uint32_t
+NetSecurity_ImportPrincipalName(uint32_t* minorStatus, char* inputName, uint32_t inputNameLen, gss_name_t* outputName)
+{
+    assert(minorStatus != nullptr);
+    assert(inputNameLen >= 0);
+    assert(outputName != nullptr);
+    gss_buffer_desc inputNameBuffer{.length = inputNameLen, .value = inputName};
+    gss_OID nameType = const_cast<gss_OID>(GSS_KRB5_NT_PRINCIPAL_NAME);
+    return gss_import_name(minorStatus, &inputNameBuffer, nameType, outputName);
+}
+
+extern "C" uint32_t NetSecurity_InitSecContext(uint32_t* minorStatus,
+                                               gss_cred_id_t claimantCredHandle,
+                                               gss_ctx_id_t* contextHandle,
+                                               uint32_t isNtlm,
+                                               gss_name_t targetName,
+                                               uint32_t reqFlags,
+                                               uint8_t* inputBytes,
+                                               uint32_t inputLength,
+                                               gss_buffer_t* outBufferHandle,
+                                               uint32_t* outTokenLength,
+                                               uint32_t* retFlags)
+{
+    assert(isNtlm == 0 || isNtlm == 1);
+    assert(minorStatus != nullptr);
+    assert(contextHandle != nullptr);
+    assert(outBufferHandle != nullptr);
+    assert(outTokenLength != nullptr);
+    assert(retFlags != nullptr);
```

Comment:
It is valid to have  null (the first time initSecContext is called, inputBytes is null).  It will point to some data only on the subsequent calls